eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -210,8 +196,7 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, watcher: ActorRe
         context unwatch d.peerConnection
         d.peerConnection ! PoisonPill
         d.channels.values.toSet[ActorRef].foreach(_ ! INPUT_DISCONNECTED) // we deduplicate with toSet because there might be two entries per channel (tmp id and final id)
```

Comment:
I'm wondering if this is the best way to handle connection switching for the underlying channels.
Currently it means the channel will go offline, cancelling incoming HTLCs and disabling the channel (with a new `channel_update`), only to re-enable it right after.

Shouldn't we go one step further and have the channel FSM handle `INPUT_RESTORED` in normal states (SYNCING, NORMAL, ...)? We wouldn't send an `INPUT_DISCONNECTED`, and the channel would simply update its `peerConnection` reference.

But maybe it's a bit ugly to have this logic bleed into the channel (and maybe we don't care much because this case happens mostly for private channels with mobile nodes)?

Reply:
We have to do it because there really is a disconnection. We could hide it but our peer will not, he will cancel htlcs and send us `channel_reestablish`. That's a protocol level thing.

Reply:
Ah right, this makes sense.